### PR TITLE
feat(markdown): support markdown grammar code generation

### PR DESCRIPTION
**Pull request by suxin2017**
## Summary

 grammer support
 support commonmark grammer defined
 
## Test Plan

add quick test

### CODE REVIEW

**Comment by suxin2017** [comment]
Can someone review it for me? @ematipico @Conaclos

File: xtask/codegen/markdown.ungram
Comment (dyc3) [inline xtask/codegen/markdown.ungram:1]: This is also missing markdown tables

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: xtask/codegen/markdown.ungram
Comment (suxin2017) [reply xtask/codegen/markdown.ungram:1]: This is the implementation of the common markdown spec so there is no table, we will add it later when we implement the gfm spec.

File: xtask/codegen/markdown.ungram [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+// indented code blocks & fenced code blocks
+AnyCodeBlock = MarkdownIndentCodeBlock | MarkdownFencedCodeBlock
+
+// [    ] xxx
+MarkdownIndentCodeBlock = MarkdownTextual
+// ~~~ 
+// xxx
+// ~~~
+MarkdownFencedCodeBlock = MarkdownTextual
```
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram:85]: Currently we are only implementing the base spec

File: xtask/codegen/markdown.ungram [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+// h1..h6
+AnyMarkdownHeader = MarkdownH1 | MarkdownH2 | MarkdownH3| MarkdownH4| MarkdownH5| MarkdownH6
+
+MarkdownH1 = MarkdownParagraph?
+MarkdownH2 =  MarkdownParagraph?
+MarkdownH3 =  MarkdownParagraph?
+MarkdownH4 =  MarkdownParagraph?
+MarkdownH5 =  MarkdownParagraph?
+MarkdownH6 =  MarkdownParagraph?
```
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram]: Thanks your review，i‘ve change it

File: xtask/codegen/markdown.ungram
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram:1]: I think it's a good idea, is it ok if I change it in the next pr

File: xtask/codegen/markdown.ungram [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+// setext title
+// foo
+// ===
+// bar
+// ---
+AnyMakrdownSetextHeader = MarkdownSetextH1 | MarkdownSetextH2
```
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram]: done

File: xtask/codegen/markdown.ungram [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+// [    ] xxx
+MarkdownIndentCodeBlock = MarkdownTextual
+// ~~~ 
+// xxx
+// ~~~
+MarkdownFencedCodeBlock = MarkdownTextual
+
+// html block
+MarkdownHTMLBlock = MarkdownTextual
```
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram]: done

File: xtask/codegen/markdown.ungram [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+// and their children will need to yield nodes and tokens as well.
+// For this reason, SyntaxElement = SyntaxElement
+
+SyntaxElement = SyntaxElement
+
+MarkdownBogus = SyntaxElement*
+
+MarkdownDocument = 
+	bom: 'UNICODE_BOM'?
+    value: AnyMarkdownBlock
```
[the remark this answers is not in the document]
Comment (suxin2017) [reply xtask/codegen/markdown.ungram]: done

File: xtask/codegen/markdown.ungram
Comment (dyc3) [reply xtask/codegen/markdown.ungram:1]: Ah, sounds good.

[the remark this answers is not in the document]
Comment (dyc3) [reply xtask/codegen/markdown.ungram:1]: Sure

File: crates/biome_ungrammar/src/lexer.rs [hunk 6]
```diff
@@ -161,5 +161,5 @@ fn is_whitespace(c: char) -> bool {
     matches!(c, ' ' | '\t' | '\n')
 }
 fn is_ident_char(c: char) -> bool {
-    matches!(c, 'a'..='z' | 'A'..='Z' | '_')
+    matches!(c, 'a'..='z' | 'A'..='Z' | '_' | '0'..='9')
```
Comment (dyc3) [inline crates/biome_ungrammar/src/lexer.rs:164]: Could you revert this change since you don't need it anymore?

**Review by dyc3 — APPROVED** [review]

File: crates/biome_ungrammar/src/lexer.rs (hunk 6 above)
Comment (suxin2017) [reply crates/biome_ungrammar/src/lexer.rs]: Thanks for your comment. I corrected it
